src/parsers/chrono/chrono.js

```diff
@@ -56,6 +56,31 @@ dayPeriodsParser.extract = (text, ref, match) => {
   });
 };
 
+const currentMonthDayParser = Object.assign(
+  new chrono.Parser(),
+  {
+    pattern: () => /\b(?:on|by)\s+the\s+(\d\d?)(?:th|st|rd|nd)\b/,
```

Comment:
Adding "on the" or "by the" at the start of the expression should avoid false positives. I'm especially thinking about actions like "meet at the 4th" (implied "street")